### PR TITLE
Feature/add group store

### PR DESCRIPTION
**Pull request by wongni**
Added groupStore module

### CODE REVIEW

**Comment by mjsalinger** [comment]
@wongni This PR looks like it is done right..  I'll review tomorrow and provide you with any feedback.

In the meantime, since some commits were done with your @wkkim-se account, can you use that account to sign the Contributor License Agreeemnt?  Thanks!

File: lib/service/modules/groupStore.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        callback = group;
+        group = null;
+
+        const error = new Error('GroupStoreError');
+        error.description = 'Bad Request';
+        error.debug = 'A group entity must be supplied';
+        return callback(error);
+      }
+
+      const requestOptions = this._buildUserRequest({ useAppSecret: true });
```
Comment (mjsalinger) [inline lib/service/modules/groupStore.js]: Creating groups does not use app secret credentials, it either uses user context or mastersecret.  So the options object here can be removed.

File: lib/service/modules/groupStore.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        err.debug = 'groupId is required';
+        return callback(err);
+      }
+
+      const requestOptions = this._buildUserRequest();
+
+      requestOptions.method = 'DELETE';
+      requestOptions.url += groupId;
+
+      if (requestContext.apiVersion > 1) {
```
Comment (mjsalinger) [inline lib/service/modules/groupStore.js]: the `hard: true` option only applies to users, not groups.

Comment (wongni) [inline lib/service/modules/groupStore.js]: Fixed and pushed

File: lib/service/modules/groupStore.js (hunk 1 above)
Comment (wongni) [inline lib/service/modules/groupStore.js]: Fixed and pushed

**Comment by wongni** [comment]
Do you know when a new npm package including this change is going to be available in the npm repository?

**Comment by mjsalinger** [comment]
Will be releasing a new version (v2.0.0) either today or tomorrow.

**Pull request merged**